geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/data/KafkaDataStoreFactory.scala

```diff
@@ -215,6 +235,32 @@ object KafkaDataStoreFactory extends GeoMesaDataStoreInfo with LazyLogging {
     KafkaDataStoreParams.IndexTiers.lookupOpt(params).flatMap(parse).getOrElse(SizeSeparatedBucketIndex.DefaultTiers)
   }
 
+  /**
+   * Parse the dynamic expiry param value into a seq of pairs
+   *
+   * @param params data store params
+   * @return
+   */
+  private [data] def parseDynamicExpiry(params: java.util.Map[String, Serializable]): Seq[(String, Duration)] = {
+    lazy val key = s"Invalid property for parameter '${KafkaDataStoreParams.DynamicCacheExpiry.key}'"
+    KafkaDataStoreParams.DynamicCacheExpiry.lookupOpt(params).toSeq.flatMap { value =>
+      ConfigFactory.parseString(value).resolve().root().unwrapped().asScala.toSeq.map {
+        case (filter, exp: String) =>
+          // validate the filter, but leave it as a string so we can optimize it based on the sft later
+          try { ECQL.toFilter(filter) } catch {
+            case NonFatal(e) => throw new IOException(s"$key, expected a CQL filter but got: $filter", e)
+          }
+          val duration = try { Duration(exp) } catch {
+            case NonFatal(e) => throw new IOException(s"$key, expected a duration key $filter but got: $exp", e)
```

Comment:
Would this read better as "expected a duration _for_ key"?

Reply:
fixed